scripts/asciidoctor-text/convert-it-all

```diff
@@ -0,0 +1,74 @@
+#!/bin/python
+
+import os
+import yaml
+import sys
+import argparse
+import subprocess
+
+def process_node(node, dir="", file_list=[]):
+    currentdir = dir
+    if "Topics" in node.keys():
+        currentdir=currentdir + "/" + node['Dir']
+        for subnode in node["Topics"]:
+            file_list = process_node(subnode, dir=currentdir, file_list=file_list)
+    else:
+        file_list.append(currentdir + "/" + node['File'])
+    return file_list
+
+
+class ConvertToText(object):
+    def __init__(self):
+        parser = argparse.ArgumentParser(
+            description='This command converts the openshift-docs assemblies to plain text.',
+            usage='convert-it-all [options]')
+
+        parser.add_argument('--input-dir', '-i', required=True, help='The input directory for the openshift-docs repo')
+        parser.add_argument('--topic-map', '-t', required=True, help='The topic map file')
+        parser.add_argument('--output-dir', '-o', required=True, help='The output directory for text')
+        parser.add_argument('--attributes', '-a', help='An optional file containing attributes')
+
+        args = parser.parse_args(sys.argv[1:])
+
+        attribute_list = []
+        if args.attributes is not None:
+            attributes = yaml.load(open(args.attributes, 'r'), Loader=yaml.FullLoader)
+            for key, value in attributes.items():
+                attribute_list = attribute_list + ["-a", key + "=\"%s\"" % value]
+
+        topic_map = yaml.safe_load_all(open(args.topic_map, 'r'))
+        mega_file_list = []
+        for map in topic_map:
+            file_list = []
+            file_list = process_node(map, file_list = file_list)
+            mega_file_list = mega_file_list + file_list
+
+        output_dir = os.path.normpath(args.output_dir)
+        os.makedirs(output_dir, exist_ok=True)
+        input_dir = os.path.normpath(args.input_dir)
+        script_dir = os.path.dirname(os.path.realpath(__file__))
+
+        for filename in mega_file_list:
+            output_file = output_dir + filename + ".txt"
+            os.makedirs(os.path.dirname(os.path.realpath(output_file)), exist_ok=True)
+            input_file = input_dir + filename + ".adoc"
+            converter_file = script_dir + "/text-converter.rb"
+            print("Processing: " + input_file)
+            command = ["asciidoctor"]
+            command = command + attribute_list
+            command = command + ["-r", converter_file]
+            command = command + ["-b", "text"]
+            command = command + ["-o", output_file]
+            command = command + ["--trace"]
+            command = command + ["--quiet"]
+            command = command + [input_file]
+
+            result = subprocess.run(command)
+            if result.returncode != 0:
+                print(result)
+                print(result.stdout)
+                exit()
```

Comment:
not needed, it will exit anyway ;)

Reply:
Not sure either, it's from https://gitlab.cee.redhat.com/dmacpher/asciidoctor-text.git